geomstats/stiefel.py

```diff
@@ -0,0 +1,351 @@
+"""
+Stiefel manifold St(n,p),
+a set of all orthonormal p-frames in n-dimensional space,
+where p <= n
+"""
+
+import geomstats.backend as gs
+
+from geomstats.embedded_manifold import EmbeddedManifold
+from geomstats.riemannian_metric import RiemannianMetric
+
+TOLERANCE = 1e-6
+EPSILON = 1e-6
+
+
+def matrix_f(sq_mat, f):
+
+    sq_mat = gs.to_ndarray(sq_mat, to_ndim=3)
+
+    [eigenvalues, vectors] = gs.linalg.eig(sq_mat)
+
+    exp_eigenvalues = f(eigenvalues)
```

Comment:
is the function always the exponential? if not, this should probably be renamed.